src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -480,8 +479,8 @@ private Process LaunchNode(string msbuildLocation, string commandLineArgs)
             if (!NativeMethodsShared.IsMono)
             {
                 // Run the child process with the same host as the currently-running process.
-                exeName = GetCurrentHost();
-                commandLineArgs = "\"" + msbuildLocation + "\" " + commandLineArgs;
+                string dotnetExe = Path.Combine(FileUtilities.GetFolderAbove(exeName, 2), "dotnet.exe");
+                exeName = File.Exists(dotnetExe) ? dotnetExe : GetCurrentHost();
```

Comment:
Would it make sense to push this into `GetCurrentHost()` so we only have to do the file-exists check once?